src/http/two/BinaryPacker.js

```diff
@@ -0,0 +1,119 @@
+import { Must } from "../../misc/Gadgets";
+
+export default class BinaryPacker {
+    constructor() {
+        this._data = new Buffer("", "binary");
+    }
+
+    binToHex(text) {
+        return Buffer(text, "binary").toString("hex");
+    }
+
+    raw() {
+        return this._data.toString("binary");
+    }
+
+    consume(length) {
+        let buf = new Buffer("", "binary");
+        Must(this._data.copy(buf, 0, 0, length) === length);
+        this._data = buf;
+    }
+
+    in(data, desc) {
+        this._data += data;
+        this.put(data, data.length, desc);
+    }
+
+    put(value, size, desc) {
```

Comment:
s/put/packed/ to emphasize that this is a reporting-only, post-update method ("put" does not work well for that because its past tense is still "put").